File: CodingSamples/DotNet/Inheritance/ClassTypeTest/DemoApp/Payroll/SalesPerson.cs
namespace Payroll
{
    //SalesPerson is derived class of Employee (base class)
    class SalesPerson : Employee
    {
        //automatic property
        public double Sales { get; set; }

        public SalesPerson(int h, float r, double s) : base(h, r)
        {
            Sales = s;
        }

        //overriding virtual method of base class
        public override double Income()
        {
            double salary = base.Income();
            if(Sales >= 20000)
                salary += 0.05 * Sales;
            return salary;
        }
    }
}

File: CodingSamples/Java/Inheritance/SubclassTest/src/SalesPerson.java
package payroll;

//defining sub-class of Employee (super class)
public class SalesPerson extends Employee {

	private double sales;

	public SalesPerson(int h, float r, double s) {
		super(h, r); // calling super-class constructor
		sales = s;
	}

	public double getSales() {
		return sales;
	}

	public void setSales(double value) {
		sales = value;
	}

	//overriding method by defining a method whose name, parameter types and return type
	//match with those of a method defined in the super class.
	public double income() {
		double salary = super.income();
		if(sales >= 20000)
			salary += 0.05 * sales;
		return salary;
	}
}

